RaceTrack/Tournament/Tournament.cs

```diff
@@ -0,0 +1,61 @@
+﻿using DataRace;
+
+namespace RaceTournament
+{
+    public class TournamentResult
+    {
+        private Queue<Racer> _racers;
+        public Queue<Racer> _winners = new Queue<Racer>();
+        public Action<Racer, Racer, Racer> OnRaceCompleted;
+
+        public TournamentResult(List<Racer> racers)
+        {
+            _racers = new Queue<Racer>(racers);
+        }
+
+        public async Task RaceAsync(Racer racer1, Racer racer2)
+        {
+            var winner = await GetWinnerAsync(racer1, racer2);
+            OnRaceCompleted(racer1, racer2, winner);
+            _winners.Enqueue(winner);
+        }
+
+        public async Task<Racer> GetWinnerAsync(Racer racer1, Racer racer2)
+        {
+            await Task.Delay(2000);
+
+            return racer1.Velocity > racer2.Velocity ? racer1 : racer2;
+        }
+
+        public async Task RunTournamentAsync()
+        {
+            while (_racers.Count > 1)
+            {
+                var tasks = new List<Task>();
+
+                while (_racers.Count > 1)
+                {
+                    var firstRacer = _racers.Dequeue();
+                    var secondRacer = _racers.Dequeue();
+
+                    tasks.Add(RaceAsync(firstRacer, secondRacer));
+
+                    if (tasks.Count >= 5)
+                    {
+                        await Task.WhenAll(tasks);
+                        tasks.Clear();
+                    }
```

Comment:
This way you wait when all 5 races end. But take a look at this situation:
You've got 4 races that take 30 seconds, and one that takes 4 minutes.
With this implementation your 4 tracks will not be used for 3:30 minutes. You should reconsider the multithreading logic